turbo/stages/headerdownload/header_algos.go

```diff
@@ -571,9 +572,44 @@ func (hd *HeaderDownload) RequestMoreHeaders(currentTime uint64) (*HeaderRequest
 	return nil, penalties
 }
 
+func (hd *HeaderDownload) RequestMoreHeadersForPOS(currentTime uint64) *HeaderRequest {
+	hd.lock.Lock()
+	defer hd.lock.Unlock()
+	if hd.anchorQueue.Len() == 0 {
+		log.Trace("Empty anchor queue")
+		return nil
+	}
+	// Assemble the request
+	if hd.missingBlockNumber > 0 {
+		return &HeaderRequest{
+			Hash:    common.Hash{},
+			Number:  hd.missingBlockNumber,
+			Length:  192,
+			Skip:    0,
+			Reverse: true,
+		}
+	} else {
+		return &HeaderRequest{
+			Hash:    common.Hash{},
+			Number:  hd.nextBlockNumberRequest,
+			Length:  192,
+			Skip:    0,
+			Reverse: true,
+		}
+	}
+
+}
+
 func (hd *HeaderDownload) UpdateRetryTime(req *HeaderRequest, currentTime, timeout uint64) {
```

Comment:
If this function is every called for the `HeaderRequest` which is created by the function `RequestMoreHeadersForPOS`, the `req.Anchor.timeouts++` will panic, because those requests do not have anchor associated with them.

Reply:
yes, sorry i did not try it with the simplifications you put in